packages/pwa-kit-extension-sdk/src/react/classes/ApplicationExtension.ts

```diff
@@ -97,33 +96,28 @@ export class ApplicationExtension<
     }
 
     /**
-     * Called on the server to serialize the extension data that will be sent to the client.
+     * Called on the server to serialize the extension's routes that will be sent to the client.
      *
-     * @returns SerializedExtension - The serialized extension data.
+     * @returns SerializedRouteProps - The serialized extension data.
      * @returns null if there's no asynchronous state that needs to be serialized
      * @throws Error if the routes have not been loaded.
      */
-    public serialize(): SerializedExtension | null {
+    public serializeRoutes(): SerializedRouteProps[] | null {
```

Comment:
On naming this, how about `serializeAsyncRoutes`? To make it clearer that it may return null if there isn't any async route.